framework/resource/internal/framework/resource.go

```diff
@@ -0,0 +1,18 @@
+package framework
+
+import (
+	"context"
+)
+
+// TODO Remove this file and change imports when original framework.Resource is updated.
+
+// NOTE this is temporary code for transition period. It will be removed as
+// soon as new github.com/operatorkit/framework.Resource is created. It should
+// look like below.
+
+type Resource interface {
+	Name() string
+
+	EnsureCreated(ctx context.Context, obj interface{}) error
+	EnsureDeleted(ctx context.Context, obj interface{}) error
+}
```

Comment:
This is how I envision new resource implementation. It sits in internal so we can have old on untouched for the time being. It's used inside `retryresource` package exclusively for the transition period.